corehq/apps/locations/static/locations/js/widgets.js

```diff
@@ -76,6 +76,20 @@ hqDefine("locations/js/widgets", [
                 },
                 processResults: function (data, params) {
                     var more = (params.page || 1) * 10 < data.total;
+                    var selectedLocations = $select[0].selectedOptions;
```

Comment:
nit: I think this would be equivalent and more direct 
```
                let selectedLocations = Array.from($select[0].selectedOptions);
                if (selectedLocations.length > 0) {
                    let locIds = selectedLocations.map(option => option.value);
                    data.results.forEach(result => {
                        if (locIds.includes(result.id)) {
                            result.disabled = true;
                        }
                    });
                }
 ```

Reply:
Thanks for refactoring this whole PR!! Really need to make some time and refresh my JS knowledge because I completely forgot about forEach :/